### PR TITLE
Fixing lakectl metastore create-symlink command

### LINKED ISSUE #2728: Update the glue/hive documentation around the Athena section
It is somewhat unclear what is possible and what isn't with lakeFS and Athena.

### LINKED ISSUE #2733: create-symlink command does not use `SymlinkInputFormat`
During a refactoring the metastore clients, the input format for the symlinks was omitted.
It means that copying metastore tables to be used by Athena is currently broken.

### PR DESCRIPTION
**Pull request by itaiad200**
closes #2733 
closes #2728 

Fix `lakectl metastore create-symlink` command:
1. The input format of the created table should be `SymlinkTextInputFormat`
2. Read the configuration of hive/glue clients.

Added Athena documentation with a walkthrough example.

### CODE REVIEW

File: docs/integrations/athena.md [hunk 1]
```diff
@@ -12,12 +12,137 @@ redirect_from: ../using/athena.html
...
+```shell
+➜   aws s3 ls s3://my-bucket/my-repo-prefix/my_table/ --recursive
+2021-11-23 17:46:29          0 my-repo-prefix/my_table/symlinks/example/main/my_table/year=2021/month=11/symlink.txt
+2021-11-23 17:46:29         60 my-repo-prefix/my_table/symlinks/example/main/my_table/year=2021/month=12/symlink.txt
+2021-11-23 17:46:30         60 my-repo-prefix/my_table/symlinks/example/main/my_table/year=2022/month=1/symlink.txt
+```
+
+An example content of a symlink file:
+```text
+s3://my-bucket/my-repo-prefix/b0165d5c5b13473d8a0f460eece9eb26
```
Comment (guy-har) [inline docs/integrations/athena.md:77]: I suppose adding some more rows should make it more clear to readers not familiar with symlink files.

**Review by guy-har — APPROVED** [review]
Thanks! the documentation looks great, we really needed that.
According to the `inputFormat` parameter, I think that:
- it's confusing that empty => same as source
- Except for the symlink case, there is nothing special with the input format
I suggest passing a boolean such as `isSymlink` and setting as symlink input format only when needed.

File: docs/integrations/athena.md (hunk 1 above)
Comment (itaiad200) [reply docs/integrations/athena.md:77]: Done.

File: pkg/metastore/ms_client.go [hunk 2]
```diff
@@ -73,18 +73,19 @@ func copyDBWithTransformLocation(ctx context.Context, fromClient, toClient Clien
 	return toClient.CreateDatabase(ctx, schema)
 }
 
-func copyOrMergeWithTransformLocation(ctx context.Context, fromClient, toClient Client, fromDB, fromTable, toDB, toTable, serde string, partition []string, transformLocation func(location string) (string, error), fixSparkPlaceHolder bool) error {
+func copyOrMergeWithTransformLocation(ctx context.Context, fromClient, toClient Client, fromDB, fromTable, toDB, toTable, toInputFormat, serde string, partition []string, transformLocation func(location string) (string, error), fixSparkPlaceHolder bool) error {
```
[the remark this answers is not in the document]
Comment (itaiad200) [reply pkg/metastore/ms_client.go:76]: Agreed, done.

File: pkg/metastore/ms_client.go [hunk 3]
```diff
@@ -339,9 +340,9 @@ func getColumnDiff(ctx context.Context, fromClient, toClient ReadClient, fromDB,
 	return Diff(colsIter, toColsIter)
 }
 
-func CopyOrMergeToSymlink(ctx context.Context, client Client, fromDB, fromTable, toDB, toTable, locationPrefix string, fixSparkPlaceHolder bool) error {
+func CopyOrMergeToSymlink(ctx context.Context, fromClient, toClient Client, fromDB, fromTable, toDB, toTable, locationPrefix, toInputFormat string, fixSparkPlaceHolder bool) error {
```
[the remark this answers is not in the document]
Comment (itaiad200) [reply pkg/metastore/ms_client.go:343]: Right, done.

**Pull request merged**